src/components/GameView/CannotCounterDialog.vue

```diff
@@ -1,47 +1,62 @@
 <template>
-  <v-dialog v-model="show" persistent max-width="650">
-    <v-card v-if="oneOff" id="cannot-counter-dialog">
-      <v-card-title>Cannot Counter</v-card-title>
-      <v-card-text>
-        <span v-if="!opponentLastTwo">
-          Your opponent has played the {{ oneOff.name }} as a one-off
-          <span v-if="target"> targeting your {{ target.name }}</span>
+  <base-dialog v-if="oneOff" v-model="show" title="Cannot Counter" id="cannot-counter-dialog">
+    <template #body>
+      <div v-if="!opponentLastTwo" class="my-2">
+        Your opponent has played the 
+        <game-card-name :card-name="oneOff.name" />
+        as a one-off
+        <span v-if="target"> targeting your 
+          <game-card-name :card-name="target.name" />
         </span>
-        <span v-else>
-          Your opponent has played {{ opponentLastTwo.name }} to Counter<span v-if="playerLastTwo">
-            your {{ playerLastTwo.name }}</span
-          >.
+      </div>
+      <div v-else class="my-2">
+        Your opponent has played 
+        <game-card-name :card-name="opponentLastTwo.name" />
+        to Counter
+        <span v-if="playerLastTwo">
+          your 
+          <game-card-name :card-name="playerLastTwo.name" />.
         </span>
-        <div class="d-flex justify-center align-center my-8">
-          <game-card :suit="oneOff.suit" :rank="oneOff.rank" />
-          <p class="ml-8">
-            {{ oneOff.ruleText }}
-          </p>
-          <div v-if="target" id="target-wrapper">
-            <span id="target-icon-wrapper" class="d-flex justify-center align-center">
-              <v-icon id="target-icon" size="x-large" color="red" icon="mdi-target" />
-            </span>
-            <game-card :suit="target.suit" :rank="target.rank" />
-          </div>
+      </div>
+      <div class="d-flex justify-center align-center my-8">
+        <game-card :suit="oneOff.suit" :rank="oneOff.rank" />
+        <p class="ml-8">
+          {{ oneOff.ruleText }}
+        </p>
+        <div v-if="target" id="target-wrapper">
+          <span id="target-icon-wrapper" class="d-flex justify-center align-center">
+            <v-icon id="target-icon" size="x-large" color="red" icon="mdi-target" />
+          </span>
+          <game-card :suit="target.suit" :rank="target.rank" />
         </div>
-        You cannot Counter, because {{ reason }}.
-      </v-card-text>
-      <v-card-actions class="d-flex justify-end">
-        <v-btn data-cy="cannot-counter-resolve" color="primary" variant="flat" @click="$emit('resolve')">
-          Resolve
-        </v-btn>
-      </v-card-actions>
-    </v-card>
-  </v-dialog>
+      </div>
+      You cannot Counter, because {{ reason }}.
+    </template>
+
+    <template #actions>
+      <v-btn
+        data-cy="cannot-counter-resolve"
+        color="surface-1"
+        variant="flat" 
+        @click="$emit('resolve')"
+      >
+        Resolve
+      </v-btn>
+    </template>
```

Comment:
I don't hate passing buttons in this way but does lead to being able to pass in any elements which might not be desired. It might be better to eventually come up with a structure of data to pass to generate the buttons instead.

Reply:
That's an interesting point. I think it's nice to err on the side of flexibility for now and keep an eye out for how to put this on rails in the future. Maybe a prop based setup for structuring the buttons would make sense. But atm this leaves it up to the consumer and avoids the need to bubble the events b/c you can bind the handlers directly on the buttons you put inside the slot, which I think is nice